File: lms.frontend/src/services/quizService.js
import apiClient from "./apiClient";

export const quizService = {
    getStudentQuizzes: async (studentId) => {
        try {
            const { data } = await apiClient.get(`/student/${studentId}/quizzes`);
            return data;
        } catch (error) {
            console.error("Error fetching quizzes:", error);
            return [];
        }
    },

    getQuizById: async (quizId) => {
        const { data } = await apiClient.get(`/quizzes/${quizId}`);
        return data;
    },

    submitQuiz: async (quizId, studentId, answers) => {
        const { data } = await apiClient.post(
            `/quizzes/${quizId}/submit?studentId=${studentId}`,
            answers
        );
        return data;
    },

    getStudentResults: async (studentId) => {
        const { data } = await apiClient.get(`/student/${studentId}/quiz-results`);
        return data;
    },
};

export default quizService;

File: lms.backend/src/main/java/com/lms/backend/service/QuizService.java
package com.lms.backend.service;

import com.lms.backend.model.Quiz;
import com.lms.backend.model.QuizResult;
import com.lms.backend.model.User;

import java.util.List;
import java.util.Map;

public interface QuizService {
    List<Quiz> getQuizzesForStudent(Long studentId);

    Quiz getQuizById(Long quizId);

    QuizResult submitQuiz(Long quizId, Long studentId, Map<Long, Integer> answers);

    List<QuizResult> getStudentResults(Long studentId);
}
